libraries/base/Data/Fixed.hs

```diff
@@ -177,58 +183,44 @@ convertFixed (Number n)
           e = ceiling (logBase 10 (fromInteger r) :: Double)
 convertFixed _ = pfail
 
-data E0
+-- @E n@ generalises @E2/E3/E6/E12@ to give more precise fixed-precision
+-- arithmetic: @Fixed (E 30)@ has 30 decimal places.
+data E (n :: Nat)
 
--- | @since 4.1.0.0
-instance HasResolution E0 where
-    resolution _ = 1
--- | resolution of 1, this works the same as Integer
+instance KnownNat n => HasResolution (E n) where
+    resolution _ = 10^natVal (undefined :: E n)
```

Comment:
Have you considered a `Proxy` argument instead of `undefined`?

Reply:
Can't quite figure out how to remove the [last use of `undefined`](https://github.com/ghc/ghc/pull/196/files#diff-0b6921dcd733c43fb86317d6b1c9ef25R90) in the file, I'll try some other time.

Reply:
Should be fixed now.